rules/rules.js

```diff
@@ -77,8 +77,34 @@ const getGroupsForItem = function (ruleConfig) {
     return ["gRules"];
 }
 
+/**
+ * Check whether a rule exists.
+ *
+ * @memberOf rules
+ * @private
+ * @param {String} uid the UID of the rule
+ * @returns {Boolean}
+ */
+const ruleExists = function (uid) {
+    return !(ruleRegistry.get(uid) == null);
+};
+
+/**
+ * Remove a rule when it exists. The rule will be immediately removed.
+ *
+ * @memberOf rules
+ * @param {String} uid the UID of the rule
+ */
+const removeRule = function (uid) {
```

Comment:
I would suggest returning a boolean indicating whether a rule was actually removed or not

Reply:
Two things to mention:

- Removing a rule is only possible when it was created in the same file
- Returning a boolean indicatin whether it was removed or not does not work with:
```javascript
return !(ruleRegistry.remove(uid) == null);
```
It seems that null is also returned when the rule was removed, but accroding to the [Java Doc](https://www.openhab.org/javadoc/latest/org/openhab/core/common/registry/registry#remove(K)) it should return the element that was removed. It really removes the rule, but it always returns `null`.

To return whether the rule was removed, I have to use that work-around:
```javascript
if (ruleExists(uid)) {
   log.info('Removing rule: {}', ruleRegistry.get(uid).name ? ruleRegistry.get(uid).name : uid);
   ruleRegistry.remove(uid);
   return !ruleExists(uid);
} else {
   return false;
}
```

Reply:
lgtm